runtime/lua/vim/lsp/log.lua

```diff
@@ -12,10 +12,10 @@ local log = {}
 -- Level numbers begin with 'trace' at 0
 log.levels = {
   TRACE = 0;
```

Comment:
`Levels by name` does not match the new order of log levels. Trace should be the last entry of `log.levels`.

Need a test, or a ticket, to demonstrate why these changes are needed. I'm not sure if both modules should be in sync at all.

Reply:
I think I am now understanding how this works now.

I think my misunderstanding draws from the fact that you must set the global log level to `TRACE` if you want to be able to individually control the logging levels from nvim-lspconfig configs, and that these levels are reversed.

`log.levels` has the most verbose logging level set to 0 (TRACE), whereas `protocol.MessageType` has the most verbose logging level set to 4 (Log).

Reply:
I should probably open an issue on the nvim-lspconfig repo sharing that you need to set the global log_level for it to work properly.